java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java

```diff
@@ -212,32 +214,66 @@ private static List<SymbolInformation> search(String projectName, ArrayList<Stri
         }
 
 		IJavaSearchScope scope;
+        var workspaceDirectoryLocation = JavaLanguageServerPlugin.getPreferencesManager().getPreferences().getRootPaths();
+        if (workspaceDirectoryLocation == null || workspaceDirectoryLocation.size() == 0) {
+            logInfo("unable to find workspace directory location");
+            return new ArrayList<>();
+        }
+        
         if (includedPaths != null && includedPaths.size() > 0) {
             ArrayList<IJavaElement> includedFragments = new ArrayList<IJavaElement>();
             for (IJavaProject proj : targetProjects) {
-                for (IPackageFragment fragment : proj.getPackageFragments()) {
-                    IPath fragmentPath = fragment.getPath();
-                    // if there's no file extension, it's a path to java package from source
-                    // else it is a path pointing to jar, ear, etc. we ignore deps for now
-                    if (fragmentPath.getFileExtension() == null) {
+                for (String includedPath : includedPaths) {
```

Comment:
So here we can also send `*.java` files right?

Reply:
Yeah that is the idea, it will scope the search to the module, but that is ok, your PR filters out other ones that are not that specific file. This just makes the JDT.LS search a little faster.

Reply:
To follow up before I merge and for anyone coming back

This will only scope to the package/module for instance:

if the included path is `io/konveyor/models/Ruleset.java` the scope of the search will be `io/konveyor/models` We then filter out any incidents that would be for any other java file in the model in golang code in the provider.